lib/_stream_readable.js

```diff
@@ -1088,23 +1088,39 @@ async function* createAsyncIterator(stream) {
     }
   }
 
+  const state = stream._readableState;
+
+  let error = state.errored;
+  let errorEmitted = state.errorEmitted;
+  let endEmitted = state.endEmitted;
+  let closeEmitted = state.closeEmitted;
+
   stream
     .on('readable', next)
-    .on('error', next)
-    .on('end', next)
-    .on('close', next);
+    .on('error', function(err) {
+      error = err;
+      errorEmitted = true;
+      next.call(this);
+    })
+    .on('end', function() {
+      endEmitted = true;
+      next.call(this);
+    })
+    .on('close', function() {
+      closeEmitted = true;
+      next.call(this);
+    });
```

Comment:
We need to explicitly listen to events for cases where proper stream behavior is override e.g. `IncomingMessage.destroy`.